lib/s3.js

```diff
@@ -2,11 +2,13 @@
 const fs = require('fs');
 const path = require('path');
 const mimeLookup = require('mime-types').lookup;
-module.exports.put = (s3, bucketName, fileName, callback) => {
-  const fileKey = path.basename(fileName);
+const datefmt = require('datefmt');
+
+module.exports.put = (s3, argv, fileName, callback) => {
+  const fileKey = argv.noprefix === true ? path.basename(fileName) : `${datefmt('%Y-%m-%d', new Date())}/${(+new Date)}/${path.basename(fileName)}`;
```

Comment:
you have path.basename twice.  lets just set the filekey as path.basename and then if noprefix is true, do `${datefmt()}/${fileKey}`